### PR TITLE
Add italic parser

### PR DESCRIPTION
**Pull request by patrickelectric**

### CODE REVIEW

File: include/maddy/italicparser.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+   * @return {void}
+   */
+  void
+  Parse(std::string& line) override
+  {
+    static std::vector<std::regex> res;
+    res.push_back(std::regex{"(?!.*`.*|.*<code>.*)\\*(?!.*`.*|.*<\\/code>.*)([^\\*]*)\\*(?!.*`.*|.*<\\/code>.*)"});
+    res.push_back(std::regex{"(?!.*`.*|.*<code>.*)\\_(?!.*`.*|.*<\\/code>.*)([^\\_]*)\\_(?!.*`.*|.*<\\/code>.*)"});
+    static std::string replacement = "<i>$1</i>";
+    for(const auto& re : res) {
```
Comment (progsource) [inline include/maddy/italicparser.h]:
```suggestion
    for (const auto& re : res)
    {
```

File: tests/maddy/test_maddy_italicparser.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    },
+    {
+      "some text _bla_ text testing _it_ out",
+      "some text <i>bla</i> text testing <i>it</i> out"
+    },
+  };
+
+  auto italicParser = std::make_shared<maddy::ItalicParser>();
+
+  for(auto& test : tests) {
```
Comment (progsource) [inline tests/maddy/test_maddy_italicparser.cpp]:
```suggestion
  for (auto& test : tests)
  {
```

File: include/maddy/parser.h [hunk 3]
```diff
@@ -131,6 +134,8 @@ class Parser
     this->strikeThroughParser->Parse(line);
 
     this->inlineCodeParser->Parse(line);
+
+    this->italicParser->Parse(line);
```
Comment (progsource) [inline include/maddy/parser.h]: In `ItalicParser.h` the comment says `Has to be used before the EmphasizedParser` - but here it isn't?

File: include/maddy/italicparser.h [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+ */
+class ItalicParser : public LineParser
+{
+public:
+  /**
+   * Parse
+   *
+   * From Markdown: `text **text**`
+   *
+   * To HTML: `text <strong>text</strong>`
```
Comment (progsource) [inline include/maddy/italicparser.h]: This comment should be updated.

**Comment by progsource** [comment]
Thank you for your contribution.

At the moment `*text*` is translated to `<em>text</em>`, so using now also just `*` for italic will not work.
I would prefer using `*italic text*` and then changing it in the other parser to `_em text_` which then would result in `<em>em text</em>` or something similar.

Additionally the [docs](https://github.com/progsource/maddy/blob/master/docs/definitions.md) should get an update, when the syntax is changed.

And feel free, to add yourself to the AUTHORS file.

File: include/maddy/italicparser.h [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+   * @method
+   * @param {std::string&} line The line to interpret
+   * @return {void}
+   */
+  void
+  Parse(std::string& line) override
+  {
+    static std::vector<std::regex> res;
+    res.push_back(std::regex{"(?!.*`.*|.*<code>.*)\\*(?!.*`.*|.*<\\/code>.*)([^\\*]*)\\*(?!.*`.*|.*<\\/code>.*)"});
+    res.push_back(std::regex{"(?!.*`.*|.*<code>.*)\\_(?!.*`.*|.*<\\/code>.*)([^\\_]*)\\_(?!.*`.*|.*<\\/code>.*)"});
```
Comment (progsource) [inline include/maddy/italicparser.h]: Whenever `Parse` is called, res will get 2 additional entries, so instead use an initializer list here.

File: tests/maddy/test_maddy_italicparser.cpp [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+#include "gmock/gmock.h"
+
+#include "maddy/italicparser.h"
+
+// -----------------------------------------------------------------------------
+
+TEST(MADDY_ITALICPARSER, ItReplacesMarkdownWithItalicHTML)
+{
+  struct testIt {
```
Comment (progsource) [inline tests/maddy/test_maddy_italicparser.cpp]:
```suggestion
  struct testIt
  {
```

File: tests/maddy/test_maddy_italicparser.cpp [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    std::string text;
+    std::string expected;
+  };
+
+  std::vector<testIt> tests{
```
Comment (progsource) [inline tests/maddy/test_maddy_italicparser.cpp]:
```suggestion
  std::vector<testIt> tests
  {
```

File: include/maddy/parser.h (hunk 3 above)
Comment (patrickelectric) [reply include/maddy/parser.h]: ctrl-c, ctrl-v mistakes.

File: include/maddy/italicparser.h [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+ */
+class ItalicParser : public LineParser
+{
+public:
+  /**
+   * Parse
+   *
+   * From Markdown: `text *text*`
+   *
+   * To HTML: `text <italic>text</italic>`
```
Comment (progsource) [inline include/maddy/italicparser.h]:
```suggestion
   * To HTML: `text <i>text</i>`
```

Comment (patrickelectric) [reply include/maddy/italicparser.h]: ooops, sorry for that, fixed.

**Pull request merged**